genie-agent/src/test/groovy/com/netflix/genie/agent/execution/services/GrpcAgentRegistrationServiceImplSpec.groovy

```diff
@@ -0,0 +1,215 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services
+
+import com.netflix.genie.agent.cli.ArgumentDelegates
+import com.netflix.genie.agent.execution.exceptions.AgentRegistrationException
+
+import com.netflix.genie.proto.AgentRegistrationRequest
+import com.netflix.genie.proto.AgentRegistrationResponse
+import com.netflix.genie.proto.AgentRegistrationServiceGrpc
+import io.grpc.stub.StreamObserver
+import io.grpc.testing.GrpcServerRule
+import org.junit.Rule
+import spock.lang.Ignore
+import spock.lang.Specification
+
+
+class GrpcAgentRegistrationServiceImplSpec extends Specification {
+
+    final String hostName = "agent.com"
```

Comment:
Any reason why you are using so many Java class stuff when you could just use `def`?

Reply:
Better autocomplete when writing the tests...